src/core/src/services/formly.form.builder.ts

```diff
@@ -42,6 +42,7 @@ export class FormlyFormBuilder {
   private registerFormControls(form: FormGroup | FormArray, fields: FormlyFieldConfig[], model: any, options: FormlyFormOptions) {
     fields.forEach((field, index) => {
       field.id = getFieldId(`formly_${this.formId}`, field, index);
+      field.model = model;
```

Comment:
this isn't enough, when model change from the root component we should re-assign the new model to the field in [patchModel](https://github.com/formly-js/ngx-formly/blob/master/src/core/src/components/formly.form.ts#L65)

Reply:
Updated See latest commit.